plugins/inputs/vsphere/client.go

```diff
@@ -191,3 +211,43 @@ func (c *Client) GetServerTime(ctx context.Context) (time.Time, error) {
 	}
 	return *t, nil
 }
+
+// GetMaxQueryMetrics returns the max_query_metrics setting as configured in vCenter
+func (c *Client) GetMaxQueryMetrics(ctx context.Context) (int, error) {
```

Comment:
ctx passed in isn't used, is this intended?

Reply:
Look a couple of lines down at om.Query(...). The context is used there.

Reply:
isn't that the new context created on line 217 from a `context.Background()` though

Reply:
Oh! Yeah, I'm supposed to pass ctx into that call. The idea was to replace ctx with one that had timeout set.